### ISSUE TITLE
Artifacts

### ISSUE THREAD

**Issue by GreenMan36** [yvt/openspades#938]
I noticed this repo had GitHub actions enabled so I thought, "great I can just download and run the latest artifact to test!" but it seems I'd need to download all the requirements and compile it myself since you don't have artifacts setup to be saved.

Artifacts should be allowed to be retained for a few days, as far as I know at no cost. And they will automatically be deleted after that amount of days. This allows you (and others) to directly download and test the artifact right after merging.

**Comment by yvt** [comment]
The thing is, we want to limit the dissemination of unofficial builds. There have been times when OpenSpades accidentally included game-breaking bugs that could be considered "cheating." For official releases, server administrators could ban a specific version of OpenSpades, like aloha.pk has already done to OS 0.0.12b. Untested builds are more likely to include such bugs, and they would be much harder to control if they were distributed without any download restrictions or any means for the server administrators and the server programs to identify them.

I think the following things should happen before making the artifacts available:

- There should be some hoops to go through, such as needing to discover our GitHub Actions page. I just found out that people would [need a GitHub account](https://github.community/t/public-read-access-to-actions-artifacts/17363/11) to access the artifacts, which could be good enough.
- Automated builds should be clearly indicated to the users as such, e.g., by displaying an alternative banner in the startup window.
- The version information packets sent to the game servers should somehow indicate that the player is using an unofficial build. A versioning scheme could be established, e.g., by reserving odd patch numbers (0.1.5, 0.1.7, ...) for such builds.
